app/src/main/java/com/asfoundation/wallet/ui/onboarding/OnboardingPageChangeListener.kt

```diff
@@ -121,13 +86,73 @@ class OnboardingPageChangeListener internal constructor(private val view: View)
   override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
     lottieView!!.progress =
         position * (1f / ANIMATION_TRANSITIONS) + positionOffset * (1f / ANIMATION_TRANSITIONS)
-    checkBox!!.setOnClickListener {
-      showWarningText(position)
-      showButton(position)
-    }
-    showWarningText(position)
-    showButton(position)
+    checkBox!!.setOnClickListener { handleUI(position) }
```

Comment:
I doubt that this can be null, but maybe it can, but at least replace the !! for ?